enhanced/scraper.go

```diff
@@ -59,60 +59,72 @@ func (s *scraper) start(ctx context.Context, interval time.Duration, ch chan<- m
 
 // scrape performs a single scrape.
 func (s *scraper) scrape(ctx context.Context) map[string][]prometheus.Metric {
-	input := &cloudwatchlogs.FilterLogEventsInput{
-		LogGroupName:   aws.String("RDSOSMetrics"),
-		LogStreamNames: aws.StringSlice(s.logStreamNames),
-		StartTime:      aws.Int64(aws.TimeUnixMilli(s.nextStartTime)),
-	}
-	s.logger.Debugf("Requesting metrics since %s (last %s).", s.nextStartTime.UTC(), time.Since(s.nextStartTime))
 
-	// collect all returned events and metrics
 	allMetrics := make(map[string]map[time.Time][]prometheus.Metric) // ResourceID -> event timestamp -> metrics
-	collectAllMetrics := func(output *cloudwatchlogs.FilterLogEventsOutput, lastPage bool) bool {
-		for _, event := range output.Events {
-			l := s.logger.With("EventId", *event.EventId).With("LogStreamName", *event.LogStreamName)
-			l = l.With("Timestamp", aws.MillisecondsTimeValue(event.Timestamp).UTC())
-			l = l.With("IngestionTime", aws.MillisecondsTimeValue(event.IngestionTime).UTC())
-
-			var instance *sessions.Instance
-			for _, i := range s.instances {
-				if i.ResourceID == *event.LogStreamName {
-					instance = &i
-					break
+
+	// LogStreamNames parameter supports up to 100 items.
+	// https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_FilterLogEvents.html
+	for i := 0; len(s.logStreamNames)-(100*i) > 0; i++ {
+		slice_start := i * 100
+		slice_end := slice_start + 100
+		if slice_end > len(s.logStreamNames) {
+			slice_end = len(s.logStreamNames)
+		}
+
+		input := &cloudwatchlogs.FilterLogEventsInput{
+			LogGroupName:   aws.String("RDSOSMetrics"),
+			LogStreamNames: aws.StringSlice(s.logStreamNames[slice_start:slice_end]),
+			StartTime:      aws.Int64(aws.TimeUnixMilli(s.nextStartTime)),
+		}
+		s.logger.Debugf("Requesting metrics since %s (last %s).", s.nextStartTime.UTC(), time.Since(s.nextStartTime))
```

Comment:
You can use the rich-logging of fields instead: `s.logger.With("next_start", s.nextStartTime).With("since_last", time.Since(s.nextStartTime)).Debugf("requesting metrics")`

Rather than composing a `printf` string yourself.